core/controllers/retrieve.py

```diff
@@ -0,0 +1,42 @@
+from flask import abort, make_response, request
+from sqlalchemy import select
+
+from core.const import EXCEL_MIMETYPE
+from core.db import db
+from core.db.entities import Programme, ProgrammeProgress, Submission
+
+
+def retrieve():
+    """Handle the download request and return the originally submitted spreadsheet.
+
+    Select file by:
+    - reporting round
+    - programme name
+
+    :return: Flask response object containing the requested spreadsheet.
+    """
+    reporting_round = request.args.get("reporting_round")
+    programme_name = request.args.get("programme_name")
+
+    query = (
+        select(Submission.submission_file)
+        .join(ProgrammeProgress, Submission.id == ProgrammeProgress.submission_id)
+        .join(Programme, ProgrammeProgress.programme_id == Programme.id)
+        .where(Programme.programme_name == programme_name)
+        .where(Submission.reporting_round == reporting_round)
+    )
```

Comment:
I suppose we have to join to a programme-level table with our current data model... 

We could leave a comment to point this out as it's arbitrary which one we join to and is not clear on the face of it?

Reply:
I spent some time trying to think this one through. I think the answer might be to select files by submission_id instead, this might simplify the process.

Reply:
it does seem like we could have a programme_id field in the submission table to allow this kind of look up though